pkg/register/document.go

```diff
@@ -85,6 +109,39 @@ func (document RegisterDocument) PublicKeyByID(id string) *RegisterPublicKey {
 	return nil
 }
 
+// ValidateUpdateDocument checks if the document provided is a valid update to the existing document
+func (document RegisterDocument) ValidateUpdateDocument(newDocument *RegisterDocument) error {
+	// Check Public Keys have not been removed
+	for _, key := range document.PublicKeys {
+		if err := key.ExistsInSlice(newDocument.PublicKeys); err != nil {
+			return err
```

Comment:
returning err here but not below? I see that it's adding "auth key / delegation of control ... etc " to the message but surely would be better if they all err'd the same way

Reply:
689aee0f763917eb0ca06eda79d716359a98ac44